flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/HsMemoryDataManager.java

```diff
@@ -99,6 +115,20 @@ public HsMemoryDataManager(
                             bufferCompressor,
                             this);
         }
+        this.previousPoolSize = getPoolSize();
+        if (poolSizeCheckInterval > 0) {
+            poolSizeChecker.scheduleAtFixedRate(
+                    () -> {
+                        int currentPoolSize = getPoolSize();
+                        if (previousPoolSize > currentPoolSize) {
+                            callWithLock(() -> spillStrategy.decideActionWithGlobalInfo(this));
+                        }
+                        previousPoolSize = currentPoolSize;
```

Comment:
This is unsafe. The read/write of `previousPoolSize` is not atomic.

I think it would be fine to use an AtomicInteger. The cost should not be high, as it's accessed only from the checking thread and from inside `decideActionWithGlobalInfo`.

Reply:
good catch.